Makefile

```diff
@@ -1,17 +1,31 @@
-.PHONY: vendor all clean dep
+BIN = kube-spawn
+DOCKERIZED ?= y
+DOCKER_TAG ?= latest
+PREFIX ?= /usr
+BINDIR ?= ${PREFIX}/bin
+UID=$(shell id -u)
+
+.PHONY: all clean dep
 
 VERSION=$(shell git describe --tags --always --dirty)
 
+ifeq ($(DOCKERIZED),y)
+all:
+	docker build -t kube-spawn-build:$(DOCKER_TAG) -f Dockerfile.build .
+	docker run --rm -ti \
+		-v `pwd`:/go/src/github.com/kinvolk/kube-spawn:Z \
+		--user $(UID):$(UID) \
+		kube-spawn-build
+else
 all:
 	go build -o cni-noop ./cmd/cni-noop
 	go build -o cnispawn ./cmd/cnispawn
 	go build -o kube-spawn-runc ./cmd/kube-spawn-runc
 	go build -o kube-spawn \
 		-ldflags "-X main.version=$(VERSION)" \
 		./cmd/kube-spawn
+endif
 
-vendor: | dep
```

Comment:
Should we keep this and call it `update-vendor` or similar?

Reply:
@iaguis Done